automates/text_reading/src/test/scala/org/clulab/aske/automates/text/TestFunctions.scala

```diff
@@ -96,31 +140,189 @@ class TestFunctions extends ExtractionTest {
   val t3e = "Wilting point Wp and field capacity Wc were calculated from soil depth and soil texture information, i.e., the relative proportion of sand, silt and clay, according to a set of prediction equations developed by Saxton et al. (1986)."
   failingTest should s"find functions from t3e: ${t3e}" taggedAs(Somebody) in {
     val desired = Seq(
-      "Wilting point Wp and field capacity Wc" -> Seq("soil depth", "soil texture information")
+      "Wilting point Wp" -> Seq("soil depth", "soil texture information"), // multiple outputs: needs to be addressed in the next meeting!
+      "field capacity Wc" -> Seq("soil depth", "soil texture information")
     )
     val mentions = extractMentions(t3e)
     testFunctionEvent(mentions, desired)
   }
 
-}
+  // Tests from 2003-DSSAT
 
+  val t1f = "the soil water content of each layer is updated by adding or subtracting daily flows of water to or from the layer due to each process."
+  failingTest should s"find functions from t1f: ${t1f}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "the soil water content of each layer" -> Seq("daily flows of water", "layer"),
+      "the soil water content of each layer" -> Seq("daily flows of water", "layer")
+    )
+    val mentions = extractMentions(t1f)
```

Comment:
Hm... I wonder if we should think of adding the type of relation involved in the function as an attachment or something... (Could decide based on the trigger). Like here, there are two identical-looking (from the point of view of test) extractions. If there were an attachment with the type of relation involved, they would be different and it would make more sense (that would involved some modifications to how we test these and also some additional work on attachments, but that shouldn't be too bad).

With the way we test events like this, there are two checks:
- number of extractions is correct
- every 'desired' mention exists in the extracted mentions

The first one is good, but the problem with the second on is that one of the two extracted mentions can be completely different from desired and the test will still pass (because both of the desired are identical, so there has to be only one mention with the desired arguments for the test to be like "yep, mentions does contain an event with these arguments!"). I wonder if right here we can throw another check right here (for this specific test) that would say that text of the first one of the `mentions` is equal to the second one.

Reply:
This is an excellent point! I've been thinking that function tests should check for the trigger (or type of relation) along with inputs and outputs, but wasn't sure how to fix them. Thanks for the great ideas! Let's discuss this in more detail during our next meeting.